mdn/src/main/java/org/apache/james/mdn/MDN.java

```diff
@@ -131,9 +131,9 @@ public static MDN parse(Message message) throws MDNParseException {
                     .map(report -> MDN.builder()
                         .humanReadableText(humanReadableText)
                         .report(report)
-                        .includeOriginalMessage(isIncludeOriginalMessage(bodyParts))
+                        .message(Optional.of(message))
```

Comment:
No, we should 
 - look if we have a third body part
 - validate that it have a `message/rfc822` Content-Type header